src/components/commentAdder.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import { useState} from 'react'
+import { postComment } from '../utils/api';
+
+function CommentAdder({setArticleComments, article_id}) {
+
+
+  const [newComment, setNewComment] = useState([])
+  const [username, setUsername] = useState('tickle122')
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    postComment(article_id, username, newComment)
+    .then(({data: {post}}) =>{
```

Comment:
this needs to be optimistically rendered so make sure the setting of state is independent of the api call.

Reply:
will do - thank you